test/core/test/link.js

```diff
@@ -6,399 +6,487 @@
     describe('Link', function() {
         this.timeout(35000);
 
-        before(Utils.createCKEditor);
+        describe('general behaviour', function() {
```

Comment:
This whole change is impossible to track down. I've been trying to put it in order hoping to see just some additions, but it turns out that several tests are now missing without explanation and others are changing assertions.

Please, we need to redo this. The default suite should stay mostly the same albeit specific needed changes that should be properly explained. Then, we can add an additional `describe('when appendProtocol is false')` to test the additional case.